mass_mailing_autoresponder_crm_import/wizard/import_knime_export_wizard.py

```diff
@@ -0,0 +1,191 @@
+        """Handle an individual row."""
+        partner = partner_obj.search([('email', '=', email)], limit=1)
+        if not partner:
+            raise UserError(
+                _('Can not find jobseeker with email: {}').format(email))
+        if segment == 'a' or segment == 'b' or segment == 'c':
+            partner.segment_jobseeker = segment
+        else:
+            raise UserError(_('Invalid segment. Segment should be A, B or C'))
+        campaign = campaign_obj.search(
+            [('name', 'ilike', active_email)],
+            limit=1)
+        if len(campaign) == 1:
+            partner.mail_adkd_campaign_ids = [(4, campaign.id)]
+        else:
+            campaign = campaign_obj.create({'name': active_email})
+            partner.mail_adkd_campaign_ids = [(4, campaign.id)]
+
+
+    def import_data(self):
+        """Parse and import a file."""
+        if not self.file or not \
+                self.filename.lower().endswith(
+                    ('.xls', '.xlsx', '.csv', 'txt')):
+            raise UserError(
+                _("Please Select an .xls, .xlsx, .txt or .csv file to Import.")
+            )
+        partner_obj = self.env['res.partner']
+        campaign_obj = self.env['res.partner.mail.adkd.campaign']
+
+        if self.filename.lower().endswith(('.csv', '.txt')):
+            # Decode data to string, Odoo supplies it as base64
+            csv_data = base64.b64decode(self.file)
+            data_file = io.StringIO(csv_data.decode("utf-8"))
+            # Read CSV
+            headers, *data = csv.reader(data_file, delimiter=';')
+
+            # Verify Header, Force it lowercase and make a dict
+            headers = self.check_header(headers)
+            for row in data:
+                try:
+                    email = row[headers['e-postadress']]
+                    segment = row[headers['segment']].lower().strip()
+                    send_today = row[headers['sendtoday']]
+                    active_mail = row[headers['activemail']]
+                except IndexError:
+                    # Empty or malformed row.
+                    _logger.warning(f'Could not parse the row: {row}')
+                else:
+                    self.import_row(email, segment, send_today,
+                                    active_mail, partner_obj, campaign_obj)
+
+        elif self.filename.lower().endswith(('.xls', '.xlsx')):
+            data = base64.decodestring(self.file)
+            book = open_workbook(file_contents=data or b'')
+            sheet = book.sheet_by_index(0)
+
+            # Verify Header, Force it lowercase and make a dict
+            headers = self.check_header([cell.value for cell in sheet.row(0)])
+            for row_nr in range(sheet.nrows, 1):
+                email = sheet.cell_value(row_nr, headers['e-postadress'])
+                segment = sheet.cell_value(row_nr,
+                                           headers['segment'].lower().strip())
```

Comment:
is .lower().strip() on the correct place... I think it should be outside the parenthesis so that you make the result lower and without spaces?

Reply:
Changed lower() and strip() to outside of parenthesis and fixed bugs